### PR TITLE
bouncerscript: don't require successive nightly versions outside of prod

### PR DESCRIPTION
**Pull request by bhearsum**
This is part of the fix for https://bugzilla.mozilla.org/show_bug.cgi?id=1861749, where we want to make sure we're able to run `bouncer-locations` on Try, and we won't always be guaranteed to run it every cycle.

### CODE REVIEW

File: bouncerscript/tests/test_script.py [hunk 1]
```diff
@@ -153,13 +158,26 @@ async def test_bouncer_aliases(aliases_context, mocker):
...
+            False,
+        ),
+        (
+            [
+                {"os": "win", "id": "47767", "path": "/firefox/nightly/latest-try/firefox-62.0a1.en-US.win32.installer.exe"},
+                {"os": "linux", "id": "47764", "path": "/firefox/nightly/latest-try/firefox-62.0a1.en-US.win64.installer.exe"},
+            ],
+            [
+                {"os": "win", "id": "47767", "path": "/firefox/nightly/latest-try/firefox-63.0a1.en-US.win32.installer.exe"},
+                {"os": "linux", "id": "47764", "path": "/firefox/nightly/latest-try/firefox-63.0a1.en-US.win64.installer.exe"},
```
Comment (jcristau) [inline bouncerscript/tests/test_script.py:171]: If I'm reading this right we're testing with successive versions here, but we should probably have a test that non-successive versions don't raise?

File: bouncerscript/tests/test_script.py [hunk 2]
```diff
@@ -194,6 +217,13 @@ async def toggled_get_locations_info(*args, **kwargs):
             mocker.patch.object(bscript, "get_locations_info", new=mock)
         await bouncer_locations(locations_context)
 
+        # We don't run this check unless info is being updated.
+        if updated_info:
+            if require_successive_versions and updated_info:
```
Comment (jcristau) [inline bouncerscript/tests/test_script.py]:
```suggestion
            if require_successive_versions:
```
We already know `updated_info` is true-ish here.

Comment (bhearsum) [reply bouncerscript/tests/test_script.py]: Whoops - thanks. I meant to undo that before pushing, heh.

File: bouncerscript/tests/test_script.py (hunk 1 above)
Comment (bhearsum) [reply bouncerscript/tests/test_script.py:171]: Right yes, thank you!

**Comment by bhearsum** [comment]
Holding this as a draft until I test it with dev scriptworkers.

File: bouncerscript/src/bouncerscript/constants.py [hunk 3]
```diff
@@ -70,39 +70,39 @@
 
 _BOUNCER_PATH_REGEXES_PER_PRODUCT_DEFAULT = {
     "firefox-nightly-latest": (
-        r"^(/firefox/nightly/latest-mozilla-central-l10n/firefox-\d+\.0a1\.:lang\."
+        r"^(/firefox/nightly/latest-(mozilla-central|try)-l10n/firefox-\d+\.0a1\.:lang\."
```
Comment (bhearsum) [inline bouncerscript/src/bouncerscript/constants.py:81]: This is not ideal, but given that all we do is version bump these, it's probably OK? I could be convinced to move it to the config...

**Comment by bhearsum** [comment]
Try run for this is at https://treeherder.mozilla.org/jobs?repo=try&revision=8df90d3b340d9ec18b1723847f6d89bcd0badf20

File: bouncerscript/docker.d/worker.yml [hunk 4]
```diff
@@ -2,6 +2,10 @@ work_dir: { "$eval": "WORK_DIR" }
 artifact_dir: { "$eval": "ARTIFACTS_DIR" }
 verbose: { "$eval": "VERBOSE == 'true'" }
 taskcluster_scope_prefix: { "$eval": "TASKCLUSTER_SCOPE_PREFIX" }
+require_successive_versions:
+  $if: 'ENV == "prod"'
+  then: true
+  else: false
```
Comment (jcristau) [inline bouncerscript/docker.d/worker.yml]:
```suggestion
require_successive_versions: { "$eval": "ENV == 'prod'" }
```

**Review by jcristau — APPROVED** [review]

**Pull request merged**